ch_pipeline/analysis/timing.py

```diff
@@ -45,6 +51,36 @@ def setup(self, tcorr):
             Note that the timing correction must *already* be referenced with respect
             to the times that were used to calibrate if processing stacked data.
         """
+        # Query flag database for date ranges that need timing correction
+        needs_timing_correction_flags = None
+
+        if mpiutil.rank0:
+            connect_database()
+
+            needs_timing_correction_type = df.DataFlagType.get(
+                name="needs_timing_correction"
+            )
+
+            if needs_timing_correction_type is None:
+                raise RuntimeError(
+                    "Could not find data flag type `needs_timing_correction`"
+                )
+
+            needs_timing_correction_flags = list(
+                df.DataFlag.select().where(
+                    df.DataFlag.type == needs_timing_correction_type
+                )
+            )
```

Comment:
You may want to extract the time ranges here and use those instead of the DataFlag objects. The potential issue is that if you have a DataFlag object and do `df.type` it may go off and try and do a database query to get the type object. That's fine on rank=0 as you have a DB connection, but not on the other ranks. We may make that mistake elsewhere in the pipeline.